### ISSUE TITLE
Support for Custom Data Attributes

### ISSUE THREAD

**Issue by kearnsw** [RasaHQ/rasa#5742]
**Description of Problem**:
Although Rasa supports custom components, there is no way to include additional metadata for each utterance. This is necessary for example in the case of sentiment analysis which is the primary use case given by Rasa for custom components.

**Overview of the Solution**:
Message needs to support kwargs and the Rasa JSON data reader needs to include all fields when instantiating Messages. 

**Examples** (if relevant):
```
{
  "rasa_nlu_data": {
    "common_examples": [
      {
        "intent": "happy",
        "text": "I'm happy.",
        "sentiment": 0.8
      },
      {
        "intent": "sad",
        "text": "I feel sad.",
        "sentiment": -0.6
      }
    ],
    "regex_features": [],
    "lookup_tables": [],
    "entity_synonyms": []
  }
}
```

**Blockers** (if relevant):
None

**Definition of Done**:

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @amn41 will get back to you about it soon! ✨

**Comment by amn41** [comment]
thanks @kearnsw ! will need to think about this a bit. We are reivising our training data format anyway and would have to think about how this fits in.

**Comment by kearnsw** [comment]
Resolved by #5743
